extensions/notebook/src/book/bookModel.ts

```diff
@@ -243,7 +243,8 @@ export class BookModel {
 					treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
 					isUntitled: this.openAsUntitled,
 					version: book.version,
-					parent: element
+					parent: element,
+					hierarchyId: element.book.hierarchyId ? path.join(element.book.hierarchyId, i.toString()) : i.toString()
```

Comment:
What's the case for needing to sometimes prefix with the book hierarchy ID? 

Does this ID need to be updated ever (if the item moves around between books for example)

What does this hierarchy ID look like?

I'd also suggest making a helper function for this instead of repeating it 3 times.

Reply:
every time that we move an item we reload the elements, so the hierarchyId gets calculated again

Reply:
And the other 2 questions?

Reply:
Oh sorry about this, I added the comments in the code instead. The hierarchyId will look like you stated below  `0/0/1/...`. I use the hierarchy id to identify descendants from a specific tree item, instead of traversing the table of contents.
This is to prevent dropping an item to once of its descendants.